### PR TITLE
ENH: Enhancement for spss kwargs

### LINKED ISSUE #56356: ENH: Add kwargs to pandas.read_spss so that pyreadstat keyword args can be used
### Feature Type

- [X] Adding new functionality to pandas

- [ ] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

I have recently used pandas to convert .sav files to csv using pandas.read_spss, this has typically worked without issue.
I now have a .sav file that has text which is causing a read error in pandas. 
I have been able to resolve this outside of pandas with pyreadstat by using the encoding="latin1" keyword argument

### Feature Description

Keyword arguments [here](https://ofajardo.github.io/pyreadstat_documentation/_build/html/index.html#pyreadstat.pyreadstat.read_sav) in pyreadstat docs
Something like this allowing to pass pyreadstat arguments
May need to handle output_format="dict" 
```python
def read_spss(
    path: str | Path,
    usecols: Sequence[str] | None = None,
    convert_categoricals: bool = True,
    dtype_backend: DtypeBackend | lib.NoDefault = lib.no_default,
    **kwargs,
    ) -> DataFrame:
    ...
    df, metadata = pyreadstat.read_sav(
        stringify_path(path),
        usecols=usecols,
        apply_value_formats=convert_categoricals,
        **kwargs,
    )
    ...
    return df
```

### Alternative Solutions

No alternative

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by astronights**
- [x] closes #56356
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by tinrobot2000 — APPROVED** [review]
Thank you

**Review by datapythonista — COMMENTED** [review]
Thanks for the contribution. Looks good to me, just added some minor comments.

File: pandas/io/spss.py
Comment (astronights) [reply pandas/io/spss.py]: image

Looks like mypy complains for it.

File: pandas/tests/io/test_spss.py
Comment (astronights) [reply pandas/tests/io/test_spss.py]: Resolved.

File: pandas/io/spss.py
Comment (astronights) [reply pandas/io/spss.py:50]: Same as above.

Comment (astronights) [reply pandas/io/spss.py:50]: image

Comment (datapythonista) [reply pandas/io/spss.py:50]: Can you add the `**` to the parameter name? Otherwise you are documenting this as regular parameter with the name `kwargs`, not the special `**kwargs`.

File: pandas/io/spss.py
Comment (datapythonista) [inline pandas/io/spss.py:51]: We need to add a `versionadded` marker like above. This will be added in 3.0 I guess, @lithomas1, can you confirm this is the case please?

File: pandas/io/spss.py
Comment (datapythonista) [reply pandas/io/spss.py:27]: @Dr-Irv do you mind having a look at this please? I'm confused on what's going on.

Comment (Dr-Irv) [reply pandas/io/spss.py:27]: There's 2 options for typing `kwargs`:
```python
    **kwargs:  Any
```
or
```python
     kwargs: dict[str, Any]
```
The above is mixing the two, so pick one.

File: pandas/io/spss.py
Comment (Dr-Irv) [inline pandas/io/spss.py:30]: This is still wrong.  You either use:
```python
     kwargs: dict[str, Any]
```
or
```
    **kwargs: Any
```

Comment (astronights) [reply pandas/io/spss.py:30]: Ah apologies, let me resolve it.

File: pandas/io/spss.py
Comment (astronights) [reply pandas/io/spss.py:54]: I've added v3.

File: pandas/io/spss.py
Comment (astronights) [reply pandas/io/spss.py]: Done.

File: pandas/io/spss.py
Comment (datapythonista) [inline pandas/io/spss.py:53]:
```suggestion
    **kwargs
```

**Review by datapythonista — COMMENTED** [review]
One small thing, and also the `versionadded` needs a blank line before I think, you can see here how it is not rendering: https://pandas.pydata.org/preview/56434/docs/reference/api/pandas.read_spss.html#pandas-read-spss

File: pandas/io/spss.py
Comment (astronights) [reply pandas/io/spss.py]: Shows up now.

File: pandas/tests/generic/test_to_xarray.py
Comment (datapythonista) [inline pandas/tests/generic/test_to_xarray.py:44]: Are those related to this PR? Or are they intentional from a merge?

Comment (astronights) [reply pandas/tests/generic/test_to_xarray.py:44]: Not related to this PR. But tests were failing. Looks like the `dims` attribute was deprecated. 
image

Comment (datapythonista) [reply pandas/tests/generic/test_to_xarray.py:44]: I see. We should fix those in a separate PR then, we don't want to be in a situation where we may want to revert the change to the SPSS reader, and we need to also revert fixes to tests. Or to confuse developers by fixing xarray tests in a SPSS PR. Feel free to open the PR yourself if you want. Otherwise you can just keep your changes here, and we'll merge this once the tests are fixed in `main`.

Comment (astronights) [reply pandas/tests/generic/test_to_xarray.py:44]: Ah sure makes sense. Let me open another PR for it.

Comment (astronights) [reply pandas/tests/generic/test_to_xarray.py]: Looks like someone fixed it. Rebuilding without the warning.

**Review by datapythonista — APPROVED** [review]
Thanks @astronights, this looks good to me. I'll let someone else also have a look before merging.

File: doc/source/whatsnew/v2.2.0.rst
Comment (mroeschke) [inline doc/source/whatsnew/v2.2.0.rst:868]: Could you move this to 3.0.0.rst?

File: doc/source/whatsnew/v2.2.0.rst
Comment (mroeschke) [inline doc/source/whatsnew/v2.2.0.rst:869]: This shouldn't be modified

Comment (astronights) [reply doc/source/whatsnew/v2.2.0.rst]: I've reverted.

File: doc/source/whatsnew/v2.2.0.rst
Comment (astronights) [reply doc/source/whatsnew/v2.2.0.rst]: Moved. Thanks for pointing it out.

**Comment by datapythonista** [comment]
Sorry @astronights, I'm just realizing now. I don't think this is a bug. Nothing was broken, the spss connector didn't support arbitrary parameters, and now it does. This is an enhancement, not a bug. Can you please update accordingly? In particular the whatsnew note. Thank you!

**Comment by astronights** [comment]
Can we merge if all is good?

**Comment by astronights** [comment]
Hi, I've updated the PR with the main. You could reopen this.

**Comment by astronights** [comment]
Hi @mroeschke , the build is done.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @astronights

**Pull request merged**
